label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -146,17 +136,46 @@ public void deleteLabelApplicationRule(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       this.requestValidator.validateOrThrow(requestContext, request);
-      String ruleId = request.getId();
-      if (this.labelApplicationRuleStore.getObject(requestContext, ruleId).isEmpty()) {
-        responseObserver.onError(new StatusRuntimeException(Status.NOT_FOUND));
-        return;
-      }
-      ;
-      this.labelApplicationRuleStore.deleteObject(requestContext, ruleId);
+      this.labelApplicationRuleStore
+          .getObject(requestContext, request.getId())
+          .orElseThrow(Status.NOT_FOUND::asRuntimeException);
+      this.labelApplicationRuleStore.deleteObject(requestContext, request.getId());
       responseObserver.onNext(DeleteLabelApplicationRuleResponse.getDefaultInstance());
       responseObserver.onCompleted();
     } catch (Exception e) {
       responseObserver.onError(e);
     }
   }
+
+  private static class LabelApplicationRuleStore
```

Comment:
Why choose to make this private static? It's not a problem per se, but more files isn't always a bad thing :)